core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -182,6 +183,27 @@ class LogCleaner(initialConfig: CleanerConfig,
     cleanerManager.removeMetrics()
```

Comment:
I reckon the metrics in `LogCleanerManager` would remained removed? Perhaps they shouldn't be removed while reconfiguring?

Reply:
Yes, the metrics remained removed after reconfiguring, would it be a good idea to remove `cleanerManager.removeMetrics()` this line?

Reply:
> I reckon the metrics in LogCleanerManager would remained removed? Perhaps they shouldn't be removed while reconfiguring?

@gaurav-narula that is a good point. 

@chiacyu Could you write ut to verify that metrics of `LogCleanerManager` are not removed after reconfiguration. Also, @gaurav-narula's suggestion is a better way I feel. Could you please give a try?

Reply:
I thought `testMetricsActiveAfterReconfiguration()` already test that the metrics are not removed after reconfiguration.
We can removed the `cleanerManager.removeMetrics()` then we should also removed [this](https://github.com/apache/kafka/blob/fafa3c76dc93f3258b2cea49dfd1dc7a724a213c/core/src/test/scala/unit/kafka/log/LogCleanerTest.scala#L96) in the `testRemoveMetricsOnClose()`.

Reply:
> We can removed the cleanerManager.removeMetrics() then we should also removed [this](https://github.com/apache/kafka/blob/fafa3c76dc93f3258b2cea49dfd1dc7a724a213c/core/src/test/scala/unit/kafka/log/LogCleanerTest.scala#L96) in the testRemoveMetricsOnClose().

My point was - we don't need to re-create the metrics after reconfiguration - i.e  in reconfiguration we SHOULD NOT remove the metrics. For example,
```scala
  private[this] def shutdownCleaners(): Unit = {
    info("Shutting down the log cleaner.")
    cleaners.foreach(_.shutdown())
    cleaners.clear()
  }
  
  /**
   * Stop the background cleaner threads
   */
  def shutdown(): Unit = {
    try shutdownCleaners()
    finally removeMetrics()
    info("Shutting down the log cleaner.")
  }
```

With above changes, in `reconfigure` we call `shutdownCleaners` instead of `shutdown`. WDYT?

Reply:
Oh, I got your point now. That make sense to me, will apply the change, please take a look. Thanks.